toolchain/sem_ir/type_info.h

```diff
@@ -227,6 +227,38 @@ struct NumericTypeLiteralInfo {
   IntId bit_width_id;
 };
 
+// Information about a literal that corresponds to a type.
+struct TypeLiteralInfo {
+  enum Kind : char {
+    None,
+    // A numeric type literal such as `i8`; see `numeric` field for details.
+    Numeric,
+    // `char` / `Core.Char`.
+    Char,
+    // `str` / `Core.String`.
+    // TODO: Rename `Core.String` to `Core.Str`.
+    Str,
+  };
+
+  // Returns the type literal that would evaluate to this class type, if any.
+  static auto ForType(const File& file, ClassType class_type)
+      -> TypeLiteralInfo;
+
+  // Prints the type literal that corresponds to this type.
```

Comment:
Maybe add a comment to note the crash?
```suggestion
  // Prints the type literal that corresponds to this type. Requires a valid literal.
```

Reply:
Thinking further, maybe it'd be better to change the implementation to not crash, so that bugs in in semir don't lead to crashes while printing?